tests/test_bosonic_solver.py

```diff
@@ -3,23 +3,157 @@
 """
 import numpy as np
 from numpy.linalg import eigvalsh
+from scipy.integrate import quad
+from qutip import (Qobj, QobjEvo, sigmaz, sigmax, basis, expect, Options, destroy, basis)
+from bofin.heom import add_at_idx
+from bofin.heom import HSolverDL
+from bofin.heom import BosonicHEOMSolver
+from bofin.heom import _heom_state_dictionaries, _check_Hsys
+import pytest
+from math import sqrt, factorial
 
+def test_add_at_idx():
+    """
+    Tests the function to add at hierarchy index.
+    """
+    seq = (2, 3, 4)
+    assert add_at_idx(seq, 2, 1), (2, 3, 5)
```

Comment:
```suggestion
    assert add_at_idx(seq, 2, 1) == (2, 3, 5)
```
The syntax of assert is `assert <bool>, <error_msg>` so the current code asserts that `add_at_idx` returns a non-empty sequence and `(2, 3, 5)` is just the error message.